### PR TITLE
dev-java/jrobin: EAPI 8, min java 1.8:*

### PR DESCRIPTION
**Pull request by vaukai**
Package-Manager: Portage-3.0.30, Repoman-3.0.3
Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

### CODE REVIEW

File: dev-java/jrobin/jrobin-1.6.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+LICENSE="LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+# Common dependencies
+# POM: pom.xml
+# junit:junit:4.11 -> >=dev-java/junit-4.13.2:4
+
+CP_DEPEND="
+	dev-java/junit:4
```
Comment (fordfrog) [inline dev-java/jrobin/jrobin-1.6.0-r2.ebuild:28]: i think this is wrongly declared in `pom.xml` and should not be needed for runtime:
```
 * Messages for package dev-java/jrobin-1.6.0-r2:

 * Possibly unneeded dependencies found
 * 	junit-4
```

**Comment by fordfrog** [comment]
compiling javadoc works for me only with java 8. with java 11+ i get this error:
```
 * Generating JavaDoc ...
src/main/java/org/jrobin/core/RrdNioBackend.java:27: error: package sun.nio.ch is not visible
import sun.nio.ch.DirectBuffer;
              ^
  (package sun.nio.ch is declared in module java.base, which does not export it to the unnamed module)
1 error
 * ERROR: dev-java/jrobin-1.6.0-r2::gentoo failed (compile phase):
 *   ejavadoc failed
 * 
 * Call stack:
 *     ebuild.sh, line  127:  Called src_compile
 *   environment, line 2404:  Called java-pkg-simple_src_compile
 *   environment, line  902:  Called ejavadoc '-d' 'target/api' '-encoding' 'UTF-8' '-docencoding' 'UTF-8' '-charset' 'UTF-8' '-classpath' '/usr/share/junit-4/lib/junit.jar:/usr/share/hamcrest-core-1.3/lib/hamcrest-core.jar' '-quiet' '@sources.lst'
 *   environment, line  667:  Called die
 * The specific snippet of code:
 *       javadoc ${javadoc_args} "${@}" || die "ejavadoc failed"
 * 
 * If you need support, post the output of `emerge --info '=dev-java/jrobin-1.6.0-r2::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=dev-java/jrobin-1.6.0-r2::gentoo'`.
!!! When you file a bug report, please include the following information:
GENTOO_VM=openjdk-17  CLASSPATH="" JAVA_HOME="/usr/lib64/openjdk-17"
JAVACFLAGS="-source 1.8 -target 1.8" COMPILER=""
and of course, the output of emerge --info =jrobin-1.6.0
```
maybe @Leo3418 would know how to avoid this.

**Comment by Leo3418** [comment]
I tried `JAVADOC_ARGS+=" -J--add-exports=java.base/sun.nio.ch=ALL-UNNAMED"` as suggested by <https://bugs.openjdk.java.net/browse/JDK-8179488?focusedCommentId=14074257#comment-14074257>, but it didn't work.

However, adding `JAVADOC_ARGS="-source 8"` could eliminate the Javadoc error, which is way simpler.

**Comment by fordfrog** [comment]
> I tried `JAVADOC_ARGS="-J--add-exports=java.base/sun.nio.ch=ALL-UNNAMED"` as suggested by https://bugs.openjdk.java.net/browse/JDK-8179488?focusedCommentId=14074257#comment-14074257, but it didn't work.
> 
> However, adding `JAVADOC_ARGS="-source 8"` could eliminate the Javadoc error, which is way simpler.

thanks! that looks like a nice workaround. i suggest we use this and note there that it is a workaround for the specified issue. it would be better to fix it instead, but there's still a lot to do...

**Comment by vaukai** [comment]
> > I tried `JAVADOC_ARGS="-J--add-exports=java.base/sun.nio.ch=ALL-UNNAMED"` as suggested by https://bugs.openjdk.java.net/browse/JDK-8179488?focusedCommentId=14074257#comment-14074257, but it didn't work.
> > However, adding `JAVADOC_ARGS="-source 8"` could eliminate the Javadoc error, which is way simpler.
> 
> [...] it would be better to fix it instead, but there's still a lot to do...

And the fix should be done [upstream](https://github.com/OpenNMS/jrobin/pulls).

**Comment by fordfrog** [comment]
i get a test failure now. i suppose it needs to force a locale for the tests:
```
There were 3 failures:
1) testDayOfWeekTimeSpec(org.jrobin.core.timespec.TimeParserTest)
java.lang.AssertionError: expected:<1645095600000> but was:<1645700400000>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.jrobin.core.timespec.TimeParserTest.testDayOfWeekTimeSpec(TimeParserTest.java:561)
2) testOneEntryInRrd(org.jrobin.graph.ValueAxisTest)
java.lang.AssertionError: 
  Unexpected method call ImageWorker.drawString(" 0,0", 33, 116, java.awt.Font[family=DejaVu Sans Mono,name=DejaVu Sans Mono,style=plain,size=10], java.awt.Color[r=0,g=0,b=0]):
    ImageWorker.drawString(" 0.0", lt(120), <any>, java.awt.Font[family=DejaVu Sans Mono,name=DejaVu Sans Mono,style=plain,size=10], java.awt.Color[r=0,g=0,b=0]): expected: 1, actual: 0
	at org.easymock.internal.MockInvocationHandler.invoke(MockInvocationHandler.java:44)
	at org.easymock.internal.ObjectMethodsFilter.invoke(ObjectMethodsFilter.java:94)
	at org.easymock.internal.ClassProxyFactory$MockMethodInterceptor.intercept(ClassProxyFactory.java:97)
	at org.jrobin.graph.ImageWorker$$EnhancerByCGLIB$$4bb372ab.drawString(<generated>)
	at org.jrobin.graph.ValueAxis.draw(ValueAxis.java:165)
	at org.jrobin.graph.ValueAxisTest.checkForBasicGraph(ValueAxisTest.java:373)
	at org.jrobin.graph.ValueAxisTest.testOneEntryInRrd(ValueAxisTest.java:435)
3) testBasicEmptyRrd(org.jrobin.graph.ValueAxisTest)
java.lang.AssertionError: 
  Unexpected method call ImageWorker.drawString(" 0,0", 33, 116, java.awt.Font[family=DejaVu Sans Mono,name=DejaVu Sans Mono,style=plain,size=10], java.awt.Color[r=0,g=0,b=0]):
    ImageWorker.drawString(" 0.0", lt(120), <any>, java.awt.Font[family=DejaVu Sans Mono,name=DejaVu Sans Mono,style=plain,size=10], java.awt.Color[r=0,g=0,b=0]): expected: 1, actual: 0
	at org.easymock.internal.MockInvocationHandler.invoke(MockInvocationHandler.java:44)
	at org.easymock.internal.ObjectMethodsFilter.invoke(ObjectMethodsFilter.java:94)
	at org.easymock.internal.ClassProxyFactory$MockMethodInterceptor.intercept(ClassProxyFactory.java:97)
	at org.jrobin.graph.ImageWorker$$EnhancerByCGLIB$$4bb372ab.drawString(<generated>)
	at org.jrobin.graph.ValueAxis.draw(ValueAxis.java:165)
	at org.jrobin.graph.ValueAxisTest.checkForBasicGraph(ValueAxisTest.java:373)
	at org.jrobin.graph.ValueAxisTest.testBasicEmptyRrd(ValueAxisTest.java:421)

FAILURES!!!
```

**Comment by vaukai** [comment]
> i get a test failure now. i suppose it needs to force a locale for the tests:

@fordfrog Please give me a hint how this could be done.
@Leo3418 Any help also from your side will be highly appreciated as always.

**Comment by Leo3418** [comment]
I could get all the 74 tests passing. As a matter of fact, I used `LANG="en_US.utf8"`, and had `media-fonts/dejavu` installed and all the fonts installed by it enabled via `eselect fontconfig`.

**Pull request closed without merging**